resources/qml/main.qml

```diff
@@ -86,9 +86,20 @@ ApplicationWindow {
         target: shmoose.persistence.messageController
         onSignalMessageReceived: {
             var currentChatPartner = shmoose.getCurrentChatPartner();
-            if ( applicationActive == false || currentChatPartner.localeCompare(jid) != 0 ) {
-                newMessageNotification(id, jid, message);
+            if ( applicationActive == true || currentChatPartner.localeCompare(jid) == 0 ) {
+                return; // active app/chat, do not send
             }
+            if ( shmoose.settings.ForceOffNotifications.indexOf(jid) >= 0 ) {
+                return; // notifications disabled for this jid
+            }        
+            if ( shmoose.settings.ForceOnNotifications.indexOf(jid) < 0 ) {
+                // default notification settings apply
```

Comment:
Instead of adding an additional attribute to the database and check it here, is it possible to use already existing functionality like:
readonly property bool isGroup : shmoose.rosterController.isGroup(currentChatPartner);

Reply:
Done, using the jid parameter -- this need to check the incoming message, not current chat partner. This is really better indeed, I wish I noticed this before.